accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,229 @@
+
+## Teaching Uyuni how to use cloud RMT
+
+### Context
+
+Uyuni will create vendor channels for the pay-as-you-go integration. The advantage would be that if such vendor channels had the correct channel label, then mgr-sync would link them to the appropriate products at next `mgr-sync` time, allowing CVE Audit, Product Migration and other features requiring correct product data to work correctly.
+
+Uyuni always gets all products meta information from SCC: which products exists and all repositories assigned.
+These products are only showed in the products setup wizard page if an authentication mechanism for the repositories is available, meaning users have access to it.
+
+The proposed solution will implement a new authentication mechanism on uyuni (named `cloudrmt`, for example) to deal with cloud RMT server authentication. With this approach we will be able to reuse the existing product/channel management features.
+
+### Implementation steps - Cloud RMT server access
+
+To access the cloud RMT server uyuni server needs to know is IP address (which is not registered in DNS) and trust the server certificate.
+
+Cloud RMT IP needs to be added to `/ets/hosts` on uyuni server. IP and hostname are returned by the data extraction script.
+
+Cloud RMT https certificate is also returned by the data extraction tool and on uyuni server we need to:
+  - add the certificate to folder `/etc/pki/trust/anchors/<label>`
+  - run command `update-ca-certificates`
+
+### Implementation steps - Register new pay-as-you-go repositories
+
+After we get all repositories information from the pay-as-you-go instance, for each repository we will need to:
+
+#### Find internal SUSE SCC repository ID
+
+Repositories are identified by its URL. We can extract repository endpoint from pay-as-you-go machine and find the corresponding repository in Uyuni database. Note that the cloud RMT server have a url prefix `/repo/` which we need to remove before compare.
+
+Example:
+- URL on pay-as-you-go instance: `https://host/repo/SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+- URL uyuni database (suseSccRepository table): `https://updates.suse.com/SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+
+Query to be executed: `select id from susesccrepository where url like '%SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/';`
+
+The query above will return the internal suse scc repository ID, to be used in next step.
+
+#### Insert repository authentication data
+
+- If not exists, add a new entry to table `susecredentials`, with the new authentication type `cloudrmt`, containing the repository basic authentication credentials.
+  - URL: RMT server base URL
+  - New column to save authentication header
+- Add a new entry to table `susesccrepositoryauth` for each repository, where:
+  - repo_id: the one obtained in previous step
+  - credentials_id: the one obtain from `susecredentials`
+  - source_id: null
+  - auth_type: `cloudrmt`
```

Comment:
Why adding a new column to `suseCredentials` instead of using `auth` column of `suseSCCRepositoryAuth` ?
Are the authentication header equal for all repos on the RMT server?
What happens if you add a SLES and SLES for SAP instance to Uyuni? The credential should be the same. Are the authentication headers the same as well?

Reply:
> Why adding a new column to suseCredentials instead of using auth column of suseSCCRepositoryAuth`?

I decided to re-use the same mechanism we have on the basic authentication where we pass the suseCredentials ID.
If we have all the data in there it will be much simple to implement changes in repo sync and we avoid pass too much data on the URL on the `rhncontentsource` table.
It also gives the flexibility to add another authentication mechanisms that can show up in the future.

> authentication header equal for all repos? 

At the moment yes, but it can change in future, since cloud team is working on support for product addons, and with that, addon channels can have different data.

> "What happens if you add a SLES and SLES for SAP instance to Uyuni? The credential should be the same. Are the authentication headers the same as well?" 

I have tested it with two different machines with SLES15 and both basic authentication credentials and authentication headers on AWS where different.